core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -457,4 +457,15 @@ public void writeNamedWriteable(NamedWriteable namedWriteable) throws IOExceptio
         writeString(namedWriteable.getName());
         namedWriteable.writeTo(this);
     }
+
+    /**
+     * Writes a list of {@link NamedWriteable} to the current stream, by first writing its size and then iterating over the objects
+     * in the list
+     */
+    public void writeNamedWritableList(List<? extends NamedWriteable> list) throws IOException {
```

Comment:
In write that would be okay, but to be consistent we should then also allow Collection in read and that means we need to pass in an empty Collection to the read-method. I'm not sure if this causes more trouble than necessary, you might accidentally mix orderings etc...
I would leave it for now an generalize if the need arises?